### ISSUE TITLE
User search fails if there's an invalid avatar id in one of the indexes

### ISSUE THREAD

**Issue by ThiefMaster** [indico/indico#1643]
```
[x for x in IndexesHolder().getById('name').matchUser('r', exact=False, accent_sensitive=False) if AvatarHolder().getById(x) is None]
['30020']
```

A quick workaround would be to replace `if not onlyActivated or av.isActivated():` in MaKaC/users.py with `if av and (not onlyActivated or av.isActivated()):` but we should also check why we ended up with an invalid avatar in that index.

**Comment by pferreir** [comment]
Isn't it better to just make sure a `None` value doesn't get stored in the index?

**Comment by pferreir** [comment]
Anyway, merged into v1.2 (34b371ed1ea255a2a9c1dc839be104bb88505659) as to fix the issue in servers with corrupted indices. This is all being rewritten right now anyway.
